healthcheck/client.go

```diff
@@ -0,0 +1,61 @@
+package healthcheck
+
+import (
+	"github.com/ONSdigital/go-ns/rhttp"
+	"net/http"
+	"fmt"
+)
+
+// healthcheckClient is an implementation of Client that can be used to call the healthcheck endpoint of any service
+type healthcheckClient struct {
+	Client  httpClient
+	Url     string
+	Service string
+}
+
+// httpClient has a get method. Implemented by http.Client etc.
+type httpClient interface {
+	Get(url string) (*http.Response, error)
+}
+
+// errorResponse contains details of a failed healthcheck
+type errorResponse struct {
+	service      string
+	expectedCode int
+	actualCode   int
+	uri          string
+}
+
+// NewClient creates a new Client for a service with the given name and healthcheck endpoint (url).
+// it uses rhttp.Client to make requests to the given url, reporting on any response status code != 200.
+func NewClient(service string, url string) *healthcheckClient {
```

Comment:
Consider adding a parameter for the http client, which will be the HttpClient interface type